test_data/pil/arith_improved.pil

```diff
@@ -232,23 +237,23 @@ namespace Arith(N);
 	// I have no idea where the "- 4 * prepend_zeros(p, 16)" comes from, but it was in the original constraints.
 	let product_with_p = (|| |x| |nr| product(p, x)(nr) - 4 * prepend_zeros(p, 16)(nr))();
 
-	let eq1 = (|| |nr| std::field::modulus() + product(sf, x2f)(nr) - product(sf, x1f)(nr) - y2f(nr) + y1f(nr) + product_with_p(q0f)(nr))();
```

Comment:
At some point I got an error because the number would be an integer that gets negative and cannot be converted to a field element. But if it's field elements already, then it's fine.